src/app/SubscriptionResumptionStorage.h

```diff
@@ -92,7 +92,9 @@ class SubscriptionResumptionStorage
     struct SubscriptionList
     {
         size_t mSize;
-        std::unique_ptr<SubscriptionInfo[]> mSubscriptions = std::make_unique<SubscriptionInfo[]>(CHIP_IM_MAX_NUM_SUBSCRIPTIONS);
+        std::unique_ptr<SubscriptionInfo[]> mSubscriptions;
+        SubscriptionInfo &operator[](size_t index) { return mSubscriptions[index]; }
+        const SubscriptionInfo &operator[](size_t index) const { return mSubscriptions[index]; }
```

Comment:
This doesn't help in the interface at all. It directly ties to implementation. This should be an interface and methods, rather than overload the operator[], which really casts this whole API as fundamentally directly related to flat in-memory data structures/arrays. This could not be backed by a linked list or by lazy lookup from storage layer on call.

Reply:
The storage API has been changed to iterator pattern per your suggestion.